### PR TITLE
adds support for reading sidre datastore style files

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

File: src/libs/relay/conduit_relay_io_handle.cpp [hunk 1]
```diff
@@ -921,7 +871,14 @@ IOHandle::read(const std::string &path,
...
-        m_handle->read(path, node);
+        if(path.empty())
+        {
+            m_handle->read(node);
+        }
+        else
+        {
+            m_handle->read(path, node);
+        }
     }
```
Comment (cyrush) [inline src/libs/relay/conduit_relay_io_handle.cpp:882]: resolves #570 , but we also need to add an extra unit test 

- [ ] add that test

File: src/tests/relay/t_relay_io_handle_sidre.cpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    res = utils::join_path(res,"data");
+    return utils::join_path(res,test_fname);
+}
+
+//-----------------------------------------------------------------------------
+TEST(conduit_relay_io_handle, test_sidre_with_root)
+{
+    return;
+    io::IOHandle h;
+    h.open(relay_test_data_path("braid_3d_sidre_examples.root"),
```
Comment (cyrush) [inline src/tests/relay/t_relay_io_handle_sidre.cpp:74]: these will fail on CI, want to swap out for a smaller example file

**Comment by cyrush** [comment]
resolves #559

**Comment by cyrush** [comment]
@xjrc  lots here, I don't want to burden you with a review, but FYI, this will implement the change from io_blueprint::save, to io::blueprint::save_mesh, etc

**Comment by xjrc** [comment]
Thanks for the heads up; I'll give this a review in the next couple of days. Just so I know, is this still a WIP or is it pretty much finished with your latest batch of changes?

**Comment by cyrush** [comment]
this batch of changes are finish. I still expect some rapid changes after this PR, including adding support to read meshs from sidre style hdf5 files. Again, I don't think you need to review, just wanted to keep you in the loop.

File: src/libs/relay/conduit_relay_io_handle_sidre.cpp [hunk 3]
```diff
@@ -0,0 +1,1273 @@
...
+    {
+        if(p_first == "root")
+        {
+            read_from_root(p_next,node);
+        }
+        else
+        {
+            if(!utils::string_is_integer(p_first))
+            {
+                // TODO:ERROR!
```
Comment (cyrush) [inline src/libs/relay/conduit_relay_io_handle_sidre.cpp:267]: - [ ] add error here

File: src/libs/relay/conduit_relay_io_identify_protocol.cpp [hunk 4]
```diff
@@ -143,6 +143,20 @@ identify_protocol(const std::string &path,
...
+void
+identify_file_type(const std::string &path,
+                  std::string &file_type)
+{
+    // default to unknown
+    file_type = "unknown";
+
+    // TODO: HARD WIRED!
+    file_type = "hdf5";
+
```
Comment (cyrush) [inline src/libs/relay/conduit_relay_io_identify_protocol.cpp:156]: - [ ] bring load mesh heuristic logic here

**Comment by cyrush** [comment]
- [ ] add unit test for identify protocol

**Comment by cyrush** [comment]
merging in this batch, so I can move on to reading blueprint meshes from sidre

**Pull request merged**